numpy/_core/src/multiarray/stringdtype/casts.c

```diff
@@ -100,7 +100,9 @@ string_to_string(PyArrayMethod_Context *context, char *const data[],
 
     npy_string_allocator *iallocator = NULL;
     npy_string_allocator *oallocator = NULL;
-    NpyString_acquire_allocator2(idescr, odescr, &iallocator, &oallocator);
+    npy_string_allocator **allocators[] = {&iallocator, &oallocator, NULL};
+    PyArray_StringDTypeObject *descrs[] = {idescr, odescr, NULL};
+    NpyString_acquire_allocators(allocators, descrs);
```

Comment:
Ah, `NULL` termination is nice if you build a new array!  I don't think I have ever seen the:
```
    npy_string_allocator **allocators[] = {&iallocator, &oallocator, NULL};
```
pattern before.  To some degree, it looks convenient if you unpack the allocators:
```
npy_string_allocator *allocators[2];
func(...);
npy_string+allocator *iallocator = allocators[0];
```
although I find it also a bit strange to have that extra indirection.  So, seems strange, but either choice is fine to me.

(As super strange as it seems, I think I would actually be fine with `nin, descrs` but ignoring non-strings with a type/typenum check.  Just becuase it allows `NpyString_aquire_allocators(3, context->descriptors)`.)

Anyway, also fine to keep this.  I do prefer it over the multiple version, even if it means we need to copy the descriptors around a bit unfortunately.  (I would actually be fine to keep the short-hands also, if they seem heavily used and convenient.)

---

One comment though, where changing seems better to me:  I find it a bit strange that we pass the descrs and not the allocators to the release function(s)?